rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -5134,6 +5134,244 @@ public void processPegIn_tx_no_lockable_by_surpassing_locking_cap() throws IOExc
+        verify(provider, times(1)).setReceiveHeadersLastTimestamp(anyLong());
+        Assert.assertEquals(-99, result);
+    }
+
+    @Test
+    public void receiveHeader_previus_block_not_in_storage() throws IOException, BlockStoreException {
+
+        Repository repository = mock(Repository.class);
+        StoredBlock storedBlock = mock(StoredBlock.class);
+        BtcBlockStoreWithCache btcBlockStore = mock(BtcBlockStoreWithCache.class);
+
+        BtcBlock btcBlock2 = mock(BtcBlock.class);
+        when(btcBlock2.getPrevBlockHash()).thenReturn(Sha256Hash.ZERO_HASH);
+        when(btcBlockStore.get(Sha256Hash.ZERO_HASH)).thenReturn(storedBlock);
+
+        BridgeStorageProvider provider = spy(new BridgeStorageProvider(
+                        repository,
+                        contractAddress,
+                        bridgeConstants,
+                        activationsAfterForks
+                )
+        );
+
+        BridgeSupport bridgeSupport = getBridgeSupportMockForReceiveHeader(
+                btcBlock2,
+                btcBlockStore,
+                provider,
+                storedBlock,
+                activationsAfterForks
+        );
+
+        when(btcBlockStore.get(any())).thenReturn(null);
+        int result = bridgeSupport.receiveHeader(btcBlock2);
+
+        StoredBlock storedBlock2 = storedBlock.build(btcBlock2);
+
+        // Calls put when is adding the block header. (Saves his storedBlock)
+        verify(btcBlockStore, never()).put(storedBlock2);
+        verify(provider, times(1)).setReceiveHeadersLastTimestamp(anyLong());
+        Assert.assertEquals(-3, result);
+    }
+
+    @Test
+    public void receiveHeader_block_too_old() throws IOException, BlockStoreException {
+
+        Repository repository = mock(Repository.class);
+        StoredBlock storedBlock = mock(StoredBlock.class);
+        BtcBlockStoreWithCache btcBlockStore = mock(BtcBlockStoreWithCache.class);
+
+        BtcBlock btcBlock2 = mock(BtcBlock.class);
+        when(btcBlock2.getPrevBlockHash()).thenReturn(Sha256Hash.ZERO_HASH);
+        when(btcBlockStore.get(Sha256Hash.ZERO_HASH)).thenReturn(storedBlock);
+
+        BridgeStorageProvider provider = spy(new BridgeStorageProvider(
+                        repository,
+                        contractAddress,
+                        bridgeConstants,
+                        activationsAfterForks
+                )
+        );
+
+        BridgeSupport bridgeSupport = getBridgeSupportMockForReceiveHeader(
+                btcBlock2,
+                btcBlockStore,
+                provider,
+                storedBlock,
+                activationsAfterForks
+        );
+;
+        when(storedBlock.getHeight()).thenReturn(10, 5000, 10);
+
+        int result = bridgeSupport.receiveHeader(btcBlock2);
+
+        StoredBlock storedBlock2 = storedBlock.build(btcBlock2);
+
+        // Calls put when is adding the block header. (Saves his storedBlock)
```

Comment:
This comment is repeated in all tests, check when it's valid and when it should say something else